File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";

        int card1Idx = 0;
        int card2Idx = 0;

        for (String target : goal) {
            if (card1Idx < cards1.length && target.equals(cards1[card1Idx])) {
                card1Idx++;
            } else if (card2Idx < cards2.length && target.equals(cards2[card2Idx])) {
                card2Idx++;
            } else {
                answer = "No";
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var i = 0
        var j = 0

        goal.forEach {
            if(i < cards1.size && cards1[i] == it) i++
            else if (j < cards2.size && cards2[j] == it) j++
            else return "No"
        }


        return "Yes"
    }
}